packages/ui/components/form/selectimproved/components/Options.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useCallback, useContext, useEffect, useMemo, useState } from "react";
+
+import { useKeyPress } from "@calcom/lib/hooks/useKeyPress";
+
+import { Label } from "../../inputs/Label";
+import Item from "./Item";
+import { SelectContext } from "./SelectProvider";
+import { Option } from "./type";
+
+interface OptionsProps<T extends Option> {
+  list: T[];
+  inputValue: string;
+  isMultiple: boolean;
+  selected: T | T[] | null;
+  searchBoxRef: React.RefObject<HTMLInputElement>;
+}
+
+type FlatternedOption = Option & { current: number; groupedIndex?: number };
+
+const flatternOptions = (options: Option[], groupCount?: number): FlatternedOption[] => {
+  return options.reduce((acc, option, current) => {
+    if (option.options) {
+      return [...acc, ...flatternOptions(option.options, current + (groupCount || 0))];
```

Comment:
Recursivly call this is we have options -> current + group count keeps track of where we are in the inital list 

Needed to reference label and handle keyboard navigation correctly